### DISCUSSION TITLE
Let Text widget self-layout its width and read the width in Rust?

### DISCUSSION THREAD

**Discussion by JanDiederich** [slint-ui/slint#3219] [Q&A] [answered]
I need to implement an endless Text-Scroll in Slint.

Easy solution: Put the string twice in a `Text` widget and decrease the offset `offset_x` by a regular repeating timer. If it goes less than half the text width (-width / 2) add back half the width (+= width / 2). Not resetting just to 0, that should make it seamlessly animated (worked easily with GTK).

Here is the problem: As soon as the `Text` width is bound to a property, which I can access in code, Slint stops layouting the `Text` width, its Text may even be cut off, and it's impossible to know how wide it must be to contain its text. Therefore I don't know when to reset the scroll offset `offset_x`.

How can I force Slint to layout the `Text` width and still let me read the width in Rust code?

Slint-Code:
```slint
in-out property <length> offset_x: -10px;

// Also not working if I don't define `100px`, it's 0 then.
out property <length> out_width: 100px;

Text {
   x: offset_x;
   width: out_width;
}
```

**Comment by ogoffart** [comment, marked as answer]
Give a name to the text and use a binding from the out property rather than the other way.
eg:

```
in-out property <length> offset_x: -10px;
out property <length> out_width: text.width;  // or <=> text.width

text := Text {
   x: offset_x;
}
```

**Reply by JanDiederich** [reply]
Thanks. That works great.
